### ISSUE TITLE
"config." context keys for an object property won't correctly cause menus to update

### ISSUE THREAD

**Issue by eamodio** [microsoft/vscode#93825]
@jrieken I was attempting to change the Timeline's `timeline.excludeSources` setting from an array to an object (as requested [here](https://github.com/microsoft/vscode/commit/59c57e1e899db397deb331f13e1c8c431c7dcb93#commitcomment-37761862)).

I've pushed a branch with the changes here: [eamodio/timeline-excludesources](https://github.com/microsoft/vscode/commits/eamodio/timeline-excludesources)

Here I am trying to use the `timeline.excludeSources.<source-id>` property as the `toggled` condition.
https://github.com/microsoft/vscode/blob/2d551abb57bfde80a48f2f5517b4c5fcab2380a4/src/vs/workbench/contrib/timeline/browser/timelinePane.ts#L1177

This seems to work for the initial state, but it never gets updated when the property is removed or re-added. It looks like the cause is somewhere in here:
https://github.com/microsoft/vscode/blob/db36e74f62b1b2603906d4d2c892ab12cf632a12/src/vs/platform/contextkey/browser/contextKeyService.ts#L102-L120

Where `this._values` will contain `timeline.excludeSources.<source-id>`, but the `event.affectedKeys` will only have `timeline.excludeSources`.

**Comment by jrieken** [comment]
>  but the event.affectedKeys will only have timeline.excludeSources.

That sound like an issue with the event itself, or how should this work?

**Comment by sandy081** [comment]
`event.affectedKeys` only contain the real configuration keys but not the full nested value tree.

**Comment by jrieken** [comment]
@eamodio this looks like a case for different printing/comparing logic. We cannot traverse into objects that have changed because this happens a lot but we can think of either a new context key operator that compare object-property values or use the regexp-operator with some JSON-based value printing (which is hopefully not huge strings...)

**Comment by eamodio** [comment]
Yeah, OK. I will leave the `timeline.excludeSources` as an array, but will look into adding a new context expression for object props. Thanks!
